### ISSUE TITLE
Please implement ability to leverage iRSA

### ISSUE THREAD

**Issue by jwitko** [nabsul/k8s-ecr-login-renew#45]
Is there any chance you could implement the ability for this app/chart to have the ability to leverage the [aws-pod-identity-webhook ](https://github.com/aws/amazon-eks-pod-identity-webhook) which implements [iRSA](https://docs.aws.amazon.com/eks/latest/userguide/iam-roles-for-service-accounts.html) ?  It should be pretty straight forward if you're using a relatively modern AWS SDK version.  You would also have to remove the hard requirements for the AWS credential env vars

**Comment by nabsul** [comment]
This sounds completely doable... however, I wonder if it's really that relevant to this particular tool.

The reason I ask is that the idea behind this tool is to make accessing ECR easy from non-AWS Kubernetes clusters. If you're running in EKS, don't you have better ways to get access to ECR?

**Comment by jwitko** [comment]
> If you're running in EKS
Not running in EKS.  Using a hybrid infrastructure with kubernetes running in an on-premise datacenter but leveraging the `aws-pod-identity-webhook` to provide iRSA capabilities to the service accounts within the on-premise cluster.

**Comment by nabsul** [comment]
Silly question, but have you tried using this tool without providing the environment variables? I'm using `NewSession()` to generate credentials, so it's already not hard-coded to only use environment variables.

https://docs.aws.amazon.com/sdk-for-go/api/aws/session/

**Comment by jwitko** [comment]
Yes an up to date version of the standard SDK credentials chain would definitely support iRSA through the `instancePrincipals` method.  However the issue is your helm chart https://github.com/nabsul/k8s-ecr-login-renew/blob/main/chart/templates/005-CronJob.yaml#L48-L54

**Comment by nabsul** [comment]
I think I can make a fairly safe tweak to the chart to allow you to test this: https://github.com/nabsul/k8s-ecr-login-renew/pull/53/files

**Comment by nabsul** [comment]
helm chart is published.

**Comment by nabsul** [comment]
Actually, you might also want to change the docker image to the latest one here: https://hub.docker.com/layers/nabsul/k8s-ecr-login-renew/sha-adba342/images/sha256-b55d183b333eeebaff8affdb446601b63d7c76829974f0ee487aab2a843b35db?context=explore

It has all the latest dependency updates. I'm not sure if the current official version has a new enough AWS SDK.

The downside of this image is that I haven't tested it yet. There could be a bug, especially around the namespace detection logic 🤷
